### ISSUE TITLE
[Bug] Spatial loss map titles and saved PDFs mislabeled in on_test_epoch_end when val_steps_to_log contains out-of-range steps

### ISSUE THREAD

**Issue by GiGiKoneti** [mllam/neural-lam#747]
### Describe the Bug

In `ForecasterModule.test_step` (`neural_lam/models/module.py:657-664`), `log_spatial_losses` filters `val_steps_to_log` to include only steps within the evaluation rollout horizon (`step <= spatial_loss.shape[1]`):

```python
# neural_lam/models/module.py:657-664
log_spatial_losses = spatial_loss[
    :,
    [
        step - 1
        for step in val_steps_to_log
        if step <= spatial_loss.shape[1]
    ],
]
self.spatial_loss_maps.append(log_spatial_losses)
```

However, during test epoch aggregation in `on_test_epoch_end` (`neural_lam/models/module.py:1014-1049`), the aggregated tensor `mean_spatial_loss` (which only contains the filtered subset of steps) is zipped against the **raw, unfiltered** `val_steps_to_log`:

```python
# neural_lam/models/module.py:1014-1025
loss_map_figs = [
    vis.plot_spatial_error(
        error=loss_map,
        datastore=self.datastore,
        title=f"Test loss, t={t_i} "
        f"({(self.time_step_int * t_i)} {self.time_step_unit})",
    )
    for t_i, loss_map in zip(
        val_steps_to_log,
        mean_spatial_loss,
    )
]
```

---

### Impact & Failure Modes

When `--val_steps_to_log` contains any step that exceeds the rollout length (for example, `--val_steps_to_log 100 1 5` with `--ar_steps_eval 25`):

1. **Mislabeled Figures and Saved Artifacts**:
   - In `test_step`, only steps `1` and `5` are extracted (yielding 2 spatial slices along the time dimension).
   - In `on_test_epoch_end`, `zip([100, 1, 5], [slice_step_1, slice_step_5])` pairs:
     - Step 1 spatial loss with `t_i = 100`. The figure is titled `Test loss, t=100` and written to `spatial_loss_maps/loss_t100.pdf`.
     - Step 5 spatial loss with `t_i = 1`. The figure is titled `Test loss, t=1` and written to `spatial_loss_maps/loss_t1.pdf`.
     - Step 5 is completely dropped from the output iteration.

2. **Silent Scientific Misattribution**:
   - The logged heatmaps and exported PDF loss maps display wrong forecast horizons without crashing or alerting the user.

---

### Minimal Reproducible Example

```python
import torch

val_steps_to_log = [100, 1, 5]
spatial_loss = torch.randn(2, 25, 100)  # (B, pred_steps=25, N)

# test_step filtering logic
logged_steps = [
    step - 1 for step in val_steps_to_log if step <= spatial_loss.shape[1]
]
log_spatial_losses = spatial_loss[:, logged_steps]  # shape (2, 2, 100) -> contains step 1 and step 5

mean_spatial_loss = torch.mean(log_spatial_losses, dim=0)  # shape (2, 100)

# on_test_epoch_end pairing logic
for t_i, _ in zip(val_steps_to_log, mean_spatial_loss):
    print(f"Assigned title t={t_i}")
```

**Output:**
```text
Assigned title t=100  # Actually step 1
Assigned title t=1    # Actually step 5
```

---

### Proposed Fix

Filter `val_steps_to_log` in `on_test_epoch_end` matching the condition in `test_step` before plotting and saving:

```python
valid_val_steps = [
    step for step in val_steps_to_log if step <= spatial_loss_tensor.shape[1]
]
```
and zip against `valid_val_steps` when generating `loss_map_figs` and `pdf_loss_map_figs`.

**Comment by vikramsingh920** [comment]
hii @GiGiKoneti,
I'd like to work on this issue.
Could you please assign it to me?

**Comment by GiGiKoneti** [comment]
@vikramsingh920 The proposed fix is awaiting maintainer confirmation on this thread. Once @sadamov or @joeloskarsson sign off on the approach, feel free to open a PR and I will review it.

**Comment by sadamov** [comment]
Confirmed, the zip in `on_test_epoch_end` is out of sync with the filter in `test_step`. Note it is not reachable via the CLI (`train_model` already rejects `val_steps_to_log > ar_steps_eval`), only when the module is built programmatically or loaded from a checkpoint with a shorter eval rollout.

The proposed fix does not work though: `spatial_loss_tensor.shape[1]` is the number of kept steps, not the rollout length, so `[100, 1, 5]` would still drop step 5. I suggest to store the kept steps in `test_step` (e.g. `self._test_spatial_steps = [s for s in val_steps_to_log if s <= spatial_loss.shape[1]]`) and zip against that for both the figure titles and the `loss_t{n}.pdf` filenames, plus a test that pins the filenames for an out-of-range step. Please wait for #746 to land before opening the PR.

@vikramsingh920 feel free to go ahead with that.
